internal/controller/components/kserve/kserve_controller_actions.go

```diff
@@ -16,89 +14,27 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
-	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
 	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/api/components/v1alpha1"
 	featuresv1 "github.com/opendatahub-io/opendatahub-operator/v2/api/features/v1"
-	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/status"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
 	odherrors "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/errors"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/conditions"
 	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
 func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	k, ok := rr.Instance.(*componentApi.Kserve)
-	if !ok {
-		return fmt.Errorf("resource instance %v is not a componentApi.Kserve)", rr.Instance)
-	}
-
-	rr.Conditions.MarkUnknown(status.ConditionServingAvailable)
-
-	if k.Spec.Serving.ManagementState != operatorv1.Managed {
-		rr.Conditions.MarkFalse(
-			status.ConditionServingAvailable,
-			conditions.WithSeverity(common.ConditionSeverityInfo),
-			conditions.WithReason(string(k.Spec.Serving.ManagementState)),
-			conditions.WithMessage("Serving management state is set to: %s", string(k.Spec.Serving.ManagementState)))
-
-		return nil
-	}
-
-	if rr.DSCI.Spec.ServiceMesh == nil || rr.DSCI.Spec.ServiceMesh.ManagementState != operatorv1.Managed {
-		rr.Conditions.MarkFalse(
-			status.ConditionServingAvailable,
-			conditions.WithObservedGeneration(rr.Instance.GetGeneration()),
-			conditions.WithReason(status.ServiceMeshNotConfiguredReason),
-			conditions.WithMessage(status.ServiceMeshNeedConfiguredMessage),
-		)
-
-		return ErrServiceMeshNotConfigured
-	}
-
-	var operatorsErr error
-
-	if found, err := cluster.OperatorExists(ctx, rr.Client, serviceMeshOperator); err != nil || !found {
-		if err != nil {
-			return odherrors.NewStopErrorW(err)
-		}
-
-		operatorsErr = multierror.Append(operatorsErr, ErrServiceMeshOperatorNotInstalled)
-	}
-
-	if found, err := cluster.OperatorExists(ctx, rr.Client, serverlessOperator); err != nil || !found {
+	// from 3.0 onwards, OSSMv2 is not supported anymore by KServe, ensure it is not present before proceeding
+	// having OSSMv3 present is ok, this check is specifically for OSSMv2
+	if found, err := cluster.OperatorExists(ctx, rr.Client, serviceMeshV2Operator); err != nil || found {
```

Comment:
Can you ask for more clarification from the Kserve team on why we need to check if OSSM v2 is there?